js/ui/tree_view/ui.tree_view.base.js

```diff
@@ -861,21 +878,53 @@ const TreeViewBase = HierarchicalCollectionWidget.inherit({
         $icon.hide();
     },
 
-    _renderToggleItemVisibilityIcon: function($node, node) {
-        const $icon = $('<div>')
-            .addClass(TOGGLE_ITEM_VISIBILITY_CLASS)
-            .appendTo($node);
+    _renderToggleItemVisibilityCustomIcon: function($node, node) {
+        const $expandIcon = getImageContainer(this.option('expandButtonIcon') ?? this.option('collapseButtonIcon'));
+        const $collapseIcon = getImageContainer(this.option('collapseButtonIcon') ?? this.option('expandButtonIcon'));
+
+        $expandIcon.appendTo($node).addClass('dx-custom-expander-icon');
```

Comment:
"I think it's better to separate appendTo from classes adding because it's really important moment which can be not noted now." 
What about splitting into separate lines? seems obvious to me and hard to miss this important moment:
```
$expandIcon.appendTo($node)
              .addClass('dx-custom-expander-icon');
```

Reply:
"addClass accepts an array as a parameter, we can write" - it works but qunit tests fail with systemJs error. So i will leave it as is unless you are really into this code) but i think this is a minor change. Also a may refactor this code anyway, Yulia also left a comment on this block of code

![image](https://user-images.githubusercontent.com/1498953/221361594-f6933bf1-8b78-492e-9c4c-d61b8718219d.png)